runtime/src/accounts_db.rs

```diff
@@ -6990,6 +7003,62 @@ impl AccountsDb {
                 })
                 .sum();
 
+            // subtract data.len() from accounts_data_len for all old accounts that are in the index twice
+            if pass == 0 {
+                let pubkeys_to_accounts_data_len = |pubkeys: &[Pubkey]| {
+                    let mut accounts_data_len_from_duplicates = 0;
+                    pubkeys.iter().for_each(|pubkey| {
+                        if let Some(entry) = self.accounts_index.get_account_read_entry(pubkey) {
+                            let slot_list = entry.slot_list();
+                            if slot_list.len() < 2 {
+                                return;
+                            }
+                            let mut slot_list = slot_list.clone();
+                            slot_list.sort_unstable_by(|a, b| a.0.cmp(&b.0));
+                            assert!(slot_list[0].0 < slot_list[1].0);
```

Comment:
you can get rid of this assert. It was to make sure I was sorting like I thought. Really, we should reverse a.0.cmp(&b...) and do b.0.cmp(&a) and then remove the .rev() below.
and a comment:
sort slots from greatest to least, then skip the greatest slot and sum up the data len of the lower slots. The lower slots will be masked by the greatest slot.

Reply:
actually, I'd just make this a different function.

Reply:
Done!